src/main/java/seedu/address/model/event/Event.java

```diff
@@ -0,0 +1,191 @@
+package seedu.address.model.event;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.person.Person;
+import seedu.address.model.role.Role;
+import seedu.address.model.role.RoleHandler;
+import seedu.address.model.role.exceptions.InvalidRoleException;
+
+
+
+
+
+/**
+ * Represents an Event in the address book.
+ */
+public class Event {
+    private final String name;
+    private final Set<Person> attendees;
+    private final Set<Person> vendors;
+    private final Set<Person> sponsors;
+    private final Set<Person> volunteers;
+
+    /**
+     * Constructs a {@code Event}.
+     * Each set of attendees, vendors, sponsors, and volunteers will be initialized to an empty set.
+     *
+     * @param name A valid event name.
+     */
+    public Event(String name) {
+        this.name = name;
+        this.attendees = new HashSet<>();
+        this.vendors = new HashSet<>();
+        this.sponsors = new HashSet<>();
+        this.volunteers = new HashSet<>();
+    }
```

Comment:
Nice! This is a pretty nice way to store the roles! Not sure if it is the best way in the long run, but I think for now we can stick with it.